File: packages/backend/src/common/lancedb.ts
import { connect } from '@lancedb/lancedb';

export const connectLanceDb = () => {
  return connect(
    `s3+ddb://${process.env.LANCEDB_STORAGE_BUCKET_NAME}/idp-v2?ddbTableName=${process.env.LANCEDB_LOCK_TABLE_NAME}`,
  );
};

File: packages/backend/app/lancedb.py
from functools import lru_cache

import lancedb

from app.config import get_config


@lru_cache
def get_db() -> lancedb.DBConnection:
    config = get_config()
    return lancedb.connect(
        f"s3+ddb://{config.lancedb_express_bucket_name}?ddbTableName={config.lancedb_lock_table_name}"
    )
